### PR TITLE
Add Config Option to Disable Vendor Specific Flags for Pip in pip_library

### PR DESCRIPTION
**Pull request by xinkecf35**
This PR adds a new configuration flag for the Python that allow an end user to turn off the system guessing/patch of the pip command. The change attempts to address #1450 in cases which a user is using pip/python not from the distribution and thus may not have those pip options available.

The further changes were completed:
- Adding of the boolean `DisableVendorFlags` to the Python config
- Modifies `pip_library` to not add the vendors flags if `DisableVendorFlags` is true
- Updates `config.html` in the docs to mention the option

Changes were tested by:
- Adding the `DisableVendorFlags` into `.plzconfig.local` of a test project with the path set to Pyenv
- Above done as well in the Please project itself.

Guidance/Suggestions on how to add automated tests for the configuration flag would be greatly appreciated. I am not entirely sure about the best way of doing that seeing as the only reliable way I have of doing so requires a dependency on Pyenv

### CODE REVIEW

File: src/core/config.go [hunk 1]
```diff
@@ -282,6 +282,7 @@ func DefaultConfiguration() *Configuration {
 	config.Go.InstallTool = "please_go_install"
 	config.Python.PexTool = "please_pex"
 	config.Python.DefaultInterpreter = "python3"
+	config.Python.DisableVendorFlags = false
```
Comment (Tatskaari) [inline src/core/config.go:285]: Not strictly necessary as this is the zero value anyway

**Review by Tatskaari — APPROVED** [review]

**Comment by Tatskaari** [comment]
Ah brilliant! This was on my list of things to do. Thanks for that. :)

Any reason this is a draft PR? I'd love to get it into tomorrows release.

**Comment by xinkecf35** [comment]
> Ah brilliant! This was on my list of things to do. Thanks for that. :)
> 
> Any reason this is a draft PR? I'd love to get it into tomorrows release.
> 
> Re testing: you could add an e2e test although I'm not sure how much value that would add given the nature of the change. We'd really want to have a couple different linux envs to test it on. I'd be happy to merge as is.

It’s a draft because of the testing question. I had an idea of maybe pulling down Pyenv and setting up pip that way but that feels a little heavy handed.

**Comment by Tatskaari** [comment]
Okay well if it's all good with you, I'll land this change :)

**Comment by xinkecf35** [comment]
> Okay well if it's all good with you, I'll land this change :)
> 
> Edit: looks like you need to resolve some conflicts

Resolved the conflicts with the config.html file. And yes I am good with that.

**Pull request merged**